src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -674,97 +589,96 @@ private static LoadDocumentEntity GetDocumentPages(PostedDataEntity postedData,
             // get/set parameters
             string documentGuid = postedData.guid;
             string password = string.IsNullOrEmpty(postedData.password) ? null : postedData.password;
-            if (!File.Exists(documentGuid)) 
-            { 
-                throw new GroupDocsViewerException("File not found."); 
+            var filesDirectory = globalConfiguration.Viewer.GetFilesDirectory();
+            var cachePath = Path.Combine(filesDirectory, globalConfiguration.Viewer.GetCacheFolderName());
+            var fileFolderName = Path.GetFileName(documentGuid).Replace(".", "_");
+            string fileCacheSubFolder = Path.Combine(cachePath, fileFolderName);
+            string htmlResourcePrefix = "/viewer/resources/";
+
+            if (!File.Exists(documentGuid))
+            {
+                throw new GroupDocsViewerException("File not found.");
             }
 
-            string cachePath;
-            ViewerSettings settings = GetViewerSettings(documentGuid, out cachePath);
+            IViewerCache cache = new FileViewerCache(cachePath, fileCacheSubFolder);
+            HtmlViewer htmlViewer = new HtmlViewer(documentGuid, htmlResourcePrefix, cache);
+
+            // TODO: consider another method for re-creating cache
+            System.IO.FileInfo fileInfo = htmlViewer.GetPageFile(1);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'fileInfo'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933706)